color/src/colorspace.rs

```diff
@@ -0,0 +1,287 @@
+    }
+
+    fn from_linear_srgb(src: [f32; 3]) -> [f32; 3] {
+        const LINEAR_SRGB_TO_DISPLAYP3: [[f32; 3]; 3] = [
+            [0.822_461_96, 0.177_538_04, 0.0],
+            [0.033_194_2, 0.966_805_8, 0.0],
+            [0.017_082_632, 0.072_397_44, 0.910_519_96],
+        ];
+        matmul(&LINEAR_SRGB_TO_DISPLAYP3, src).map(lin_to_srgb)
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct XyzD65;
+
+impl Colorspace for XyzD65 {
+    const IS_LINEAR: bool = true;
+
+    const CS_TAG: Option<ColorspaceTag> = Some(ColorspaceTag::XyzD65);
+
+    fn to_linear_srgb(src: [f32; 3]) -> [f32; 3] {
+        const XYZ_TO_LINEAR_SRGB: [[f32; 3]; 3] = [
+            [3.240_97, -1.537_383_2, -0.498_610_76],
+            [-0.969_243_65, 1.875_967_5, 0.041_555_06],
+            [0.055_630_08, -0.203_976_96, 1.056_971_5],
+        ];
+        matmul(&XYZ_TO_LINEAR_SRGB, src)
+    }
+
+    fn from_linear_srgb(src: [f32; 3]) -> [f32; 3] {
+        const LINEAR_SRGB_TO_XYZ: [[f32; 3]; 3] = [
+            [0.412_390_8, 0.357_584_33, 0.180_480_8],
+            [0.212_639, 0.715_168_65, 0.072_192_32],
+            [0.019_330_818, 0.119_194_78, 0.950_532_14],
+        ];
+        matmul(&LINEAR_SRGB_TO_XYZ, src)
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct Oklab;
+
+// Matrices taken from Oklab blog post, precision reduced to f32
+const OKLAB_LAB_TO_LMS: [[f32; 3]; 3] = [
+    [1.0, 0.396_337_78, 0.215_803_76],
+    [1.0, -0.105_561_346, -0.063_854_17],
+    [1.0, -0.089_484_18, -1.291_485_5],
+];
+
+const OKLAB_LMS_TO_SRGB: [[f32; 3]; 3] = [
+    [4.076_741_7, -3.307_711_6, 0.230_969_94],
+    [-1.268_438, 2.609_757_4, -0.341_319_38],
+    [-0.004_196_086_3, -0.703_418_6, 1.707_614_7],
+];
+
+const OKLAB_SRGB_TO_LMS: [[f32; 3]; 3] = [
+    [0.412_221_46, 0.536_332_55, 0.051_445_995],
+    [0.211_903_5, 0.680_699_5, 0.107_396_96],
+    [0.088_302_46, 0.281_718_85, 0.629_978_7],
+];
+
+const OKLAB_LMS_TO_LAB: [[f32; 3]; 3] = [
+    [0.210_454_26, 0.793_617_8, -0.004_072_047],
+    [1.977_998_5, -2.428_592_2, 0.450_593_7],
+    [0.025_904_037, 0.782_771_77, -0.808_675_77],
+];
+
+impl Colorspace for Oklab {
+    const CS_TAG: Option<ColorspaceTag> = Some(ColorspaceTag::Oklab);
+
+    fn to_linear_srgb(src: [f32; 3]) -> [f32; 3] {
+        let lms = matmul(&OKLAB_LAB_TO_LMS, src).map(|x| x * x * x);
+        matmul(&OKLAB_LMS_TO_SRGB, lms)
+    }
+
+    fn from_linear_srgb(src: [f32; 3]) -> [f32; 3] {
+        let lms = matmul(&OKLAB_SRGB_TO_LMS, src).map(f32::cbrt);
+        matmul(&OKLAB_LMS_TO_LAB, lms)
+    }
+
+    fn scale_chroma(src: [f32; 3], scale: f32) -> [f32; 3] {
+        [src[0], src[1] * scale, src[2] * scale]
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct Oklch;
+
+impl Colorspace for Oklch {
+    const CS_TAG: Option<ColorspaceTag> = Some(ColorspaceTag::Oklch);
+
+    const LAYOUT: ColorspaceLayout = ColorspaceLayout::HueThird;
+
+    fn from_linear_srgb(src: [f32; 3]) -> [f32; 3] {
```

Comment:
In `bevy_color`, I believe it is possible to convert between `Oklab` and `Oklch` directly. Would this be a useful operation to support?

Reply:
I was wrestling between code size and runtime efficiency. But on reflection, I think it makes sense to have a `convert<TargetCS: ColorSpace>` method with a default impl. Probably I'll do that in a separate PR to keep the next one focused on cleanup.